packages/playwright-test/src/reporters/junit.ts

```diff
@@ -133,6 +133,94 @@ class JUnitReporter implements Reporter {
     };
     entries.push(entry);
 
+    // Xray Test Management supports testcase level properties, where additional metadata may be provided
+    // some annotations are encoded as value attributes, other as cdata content
+    const properties: XMLEntry = {
+      name: 'properties',
+      children: [] as XMLEntry[]
+    };
+
+    const textValueBasedAnnotations = ['test_id', 'test_key', 'test_summary', 'requirements'];
+    for (const annotationType of textValueBasedAnnotations) {
+      const annotation = test.annotations.find(annotation => {
+        return annotation.type === annotationType;
+      });
+      if (annotation !== undefined) {
+        const property: XMLEntry = {
+          name: 'property',
+          attributes: {
+            name: annotationType,
+            value: (annotation?.description ? annotation.description : '')
+          }
+        };
+        properties.children?.push(property);
+      }
+    }
+
+    const textContentBasedAnnotations = ['test_description'];
+    for (const annotationType of textContentBasedAnnotations) {
+      const annotation = test.annotations.find(annotation => {
+        return annotation.type === annotationType;
+      });
+      if (annotation !== undefined) {
+        const property: XMLEntry = {
+          name: 'property',
+          attributes: {
+            name: annotationType
+          },
+          text: annotation.description
+        };
+        properties.children?.push(property);
+      }
+    }
+
+    // attachments are optionally embed as base64 encoded content on inner elements
+    const embedAttachments = test.annotations.find(annotation => {
+      return annotation.type === 'embed_attachments_in_report';
```

Comment:
I think it's better to have an option on the whole junit reporter, to say "embed attachments". Even in this case, we probably should not embed files, wdyt?

Reply:
Embeding attachments is in my perspective something that users will need to opt-in explicitly and should be disabled by default as it will make reports larger.
I'm not sure if I fully understood you, but please feel free to clairify/correct.
If you're suggesting to add a global configuration for the whole reporter, I agree, having it as false by default. I'm not sure if you were suggesting to remove this logic at the test level using the 'embed_attachments_in_report' annotation type that I've implemented.. or if you were suggesting having that global configuration plus this logic... or something else :)
Well, in my perspective maybe a global option for the junit reporter would be enough... not sure if users having that option as true, would want to disable this behavior in some test(s). If this makes sense, then the existing logic combined with the global option could make sense. What are your thoughts?

Reply:
I agree this should be disabled by default. I'm inclined for a single global option on the reporter for now, unless someone explicitly asks for more fine-grained control per each test. See my other comment with suggested options.

Reply:
done, with a global option for now. updated the PR